sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/percentiles.scala

```diff
@@ -155,7 +156,7 @@ abstract class PercentileBase
     }
 
     val ordering = child.dataType match {
-      case numericType: NumericType => numericType.ordering
+      case numericType: NumericType => PhysicalAtomicType(numericType).ordering
       case intervalType: YearMonthIntervalType => intervalType.ordering
```

Comment:
These interval types should also have an ordering right?

Reply:
But do we still hit this case? Or does `PhysicalDataType.ordering` match `YearMonthIntervalType` and `DayTimeIntervalType`.

Reply:
good point. `PhysicalDataType.ordering` is enough